src/Microsoft.ServiceFabric.Actors/Runtime/KvsActorStateProviderBase.cs

```diff
@@ -611,9 +624,11 @@ Task IActorStateProvider.DeleteRemindersAsync(IReadOnlyDictionary<ActorId, IRead
         /// parameter is a collection of all actor reminders contained in the actor state provider.
         /// </returns>
         /// <exception cref="OperationCanceledException">The operation was canceled.</exception>
-        Task<IActorReminderCollection> IActorStateProvider.LoadRemindersAsync(CancellationToken cancellationToken)
+        async Task<IActorReminderCollection> IActorStateProvider.LoadRemindersAsync(CancellationToken cancellationToken)
         {
-            return this.actorStateProviderHelper.ExecuteWithRetriesAsync(
+            await this.EnsureLogicalTimeManagerInitializedAsync(cancellationToken);

```

Comment:
If the primary failed over because the logical time manager is initialized and at the same time load reminders task is in progress, EnsureLogicalTimeManagerInitializedAsync would throw FabricNotPrimaryException which rolls up to RunAsync, which results in partition fault.

I advise filing a backlog item and prioritize fixing the issue, if it happens in test/prod

Reply:
Noted.
Will try to reproduce this scenario and do the appropriate fix in a separate PR.